### PR TITLE
Add mouse button bindings

### PR DESCRIPTION
**Pull request by tomas7770**
# Description

Adds support for binding mouse buttons to input actions, similar to existing implementation of keyboard/gamepad. Serialization functions are also implemented in order to support adding bindings in an asset.
Adds a new showcase to the input sample to test the bindings.

## Checklist

- [ ] Self-review changes.
- [ ] Evaluate impact on the documentation.
- [ ] Ensure test coverage.
- [ ] Write new samples.

### CODE REVIEW

File: engine/samples/input/main.cpp
Comment (RiscadoA) [inline engine/samples/input/main.cpp:141]: This is a reference to be used in the sample docs page. Don't forget to add this section to the page too! Check out `engine/samples/input/page.md`

**Review by RiscadoA — APPROVED** [review]
Thanks for working on this @tomas7770! LGTM, just needs some changes to the input sample's page before being merged :)

File: engine/samples/input/assets/sample.bind
Comment (RiscadoA) [inline engine/samples/input/assets/sample.bind:12]: I think you shouldn't add this here - I was a bit confused as to why the sample was closing when I was trying the mouse button showcase.

**Review by RiscadoA — COMMENTED** [review]
Looks great! Ty! Just one thing before merging: some of your commit messages don't follow the format - you can use `git rebase -i HEAD~7` to edit your previous 7 commits, and change each `pick` to `reword` (or just `r`) to change their messages. Most of your commits would fall under the `feat(input)` scope - for formatting one, you should use `style: fix formatting`. The commit you have which touches both sides, I would use `feat(window)` since its mostly window related code that's being changed. Usually when we can't choose a scope we try to split the commits into multiple ones or just omit the scope entirely.

Also, some tips for future PRs:
- you can press the 'auto merge' button to merge this automatically when all conversations are resolved and when the PR is approved.
- feel free to press the resolve button on reviews after pushing fixes for them.

**Comment by tomas7770** [comment]
> Looks great! Ty! Just one thing before merging: some of your commit messages don't follow the format - you can use `git rebase -i HEAD~7` to edit your previous 7 commits, and change each `pick` to `reword` (or just `r`) to change their messages. Most of your commits would fall under the `feat(input)` scope - for formatting one, you should use `style: fix formatting`. The commit you have which touches both sides, I would use `feat(window)` since its mostly window related code that's being changed. Usually when we can't choose a scope we try to split the commits into multiple ones or just omit the scope entirely.

Something like this?
```
feat(input): partially implement mouse button bindings
feat(window): add serialization for mouse buttons
fix(input): add mouse binds to input sample, to fix crash
feat(input): add mouse button showcase to input sample
style: fix formatting
feat(input): remove mouse for next-showcase in input sample
docs(input): add mouse showcase to input sample docs
```

**Pull request merged**
